seatunnel-connectors-v2/connector-pulsar/src/main/java/org/apache/seatunnel/connectors/seatunnel/pulsar/source/PulsarSource.java

```diff
@@ -213,12 +213,14 @@ private void setPartitionDiscoverer(Config config) {
         if (StringUtils.isNotBlank(topic)) {
             this.partitionDiscoverer = new TopicListDiscoverer(Arrays.asList(StringUtils.split(topic, ",")));
         }
-        String topicPattern = config.getString(TOPIC_PATTERN.key());
-        if (StringUtils.isNotBlank(topicPattern)) {
-            if (this.partitionDiscoverer != null) {
-                throw new PulsarConnectorException(SeaTunnelAPIErrorCode.OPTION_VALIDATION_FAILED, String.format("The properties '%s' and '%s' is exclusive.", TOPIC.key(), TOPIC_PATTERN.key()));
+        if (config.hasPath(TOPIC_PATTERN.key())) {
```

Comment:
If `TOPIC_PATTERN` and `TOPIC` are exclusive, you need add `exclusive(TOPIC_PATTERN, TOPIC)` in OptionRule. And you can reference https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/sink/KafkaSinkFactory.java

Reply:
The exclusive configuration of these two parameters has been added.
<img width="929" alt="image" src="https://user-images.githubusercontent.com/40714172/215929458-88913775-5c78-48ed-a594-cb8b4e41057d.png">

Reply:
back to this question. https://github.com/apache/incubator-seatunnel/pull/3989#discussion_r1090297180

Reply:
ok, redundancy judgment deleted.